packages/suite/src/storage/migrations/index.ts

```diff
@@ -566,4 +568,83 @@ export const migrate: OnUpgradeFunc<SuiteDBSchema> = async (
         // @ts-expect-error
         backendSettings.delete('trop');
     }
+
+    if (oldVersion < 37) {
+        await updateAll(transaction, 'accounts', account => {
+            // @ts-expect-error
+            if (!account.metadata?.fileName || !account.metadata?.aesKey) {
+                return account;
+            }
+            account.metadata = {
+                key: account.metadata.key,
+                1: {
+                    // @ts-expect-error
+                    fileName: account.metadata.fileName,
+                    // @ts-expect-error
+                    aesKey: account.metadata.aesKey,
+                },
+            };
+
+            return account;
+        });
+
+        await updateAll(transaction, 'devices', device => {
+            if (
+                device.metadata.status === 'enabled' &&
+                // @ts-expect-error
+                device.metadata.fileName &&
+                // @ts-expect-error
+                device.metadata.aesKey
+            ) {
+                device.metadata = {
+                    key: device.metadata.key,
+                    status: device.metadata.status,
+                    1: {
+                        // @ts-expect-error
+                        fileName: device.metadata.fileName,
+                        // @ts-expect-error
+                        aesKey: device.metadata.aesKey,
+                    },
+                };
+            }
+
+            return device;
+        });
+
+        await updateAll(transaction, 'metadata', metadata => {
+            const updatedMetadata = {
+                selectedProvider: { labels: '' },
+                providers: [],
+                enabled: metadata.enabled,
+            };
+            // @ts-expect-error
+            if (metadata.provider) {
+                // not sure about google client id I selected
+                let clientId: string;
+
+                // @ts-expect-error
+                switch (metadata.provider.type) {
+                    case 'dropbox':
+                        clientId = 'wg0yz2pbgjyhoda';
```

Comment:
I am against it, it could easily break in the future when we change this DROPBOX_CLIENT_ID. This is general problem with migration, it would be best if they relied on built and inlined code valid at the time when migration was created. You know, the problem with expect-errors everywhere because we have only latest version of types but migration works with some long abandoned shapes of data.